simple-factsheet.Rmd

```diff
@@ -23,91 +28,401 @@ library(tidyverse)
+
+states_subset <- read_excel(filepath) %>%
+  filter(StateName == params$state) %>%
+  rename_at(vars(contains("1718")),~sub("_1718", "", .)) %>%
+  
+  # prepare overall cOI population number - rounded thousand or numeric + "million" text
+  mutate(text_1_popcount = case_when(total_popcount_coi < 1000000 ~ formatC(total_popcount_coi,
+                                                                            format ="f", 
+                                                                            digits = 0,
+                                                                            big.mark =","),
+                                     TRUE ~ paste(round((total_popcount_coi/1000000),1), "million"))) %>%
+  
+  # convert decimals to be communicated as percentages - no decimals except under 1% 
+  # Except for those that will only be used in data vis 
+  mutate_at(vars(starts_with("pct"),
+                 starts_with("share")), topercent) %>%
+  
+  # prepare strings that'll change
+  mutate(text_2_citizens = case_when(share_uscitizen_coi >= 85 ~ "Almost all",
+                                     share_uscitizen_coi < 85 & share_uscitizen_coi >= 50 ~ "Most",
+                                     share_uscitizen_coi < 50 ~ "Many",
+                                     TRUE ~ NA_character_),
+         
+         text_3_citizens = case_when(share_prtnatcit_coi >= 50 ~ "most",
+                                     share_prtnatcit_coi < 50 ~ "many",
+                                     TRUE ~ NA_character_),
+         
+         temp_working_diff = share_workingfam_coi - share_workingfam_nat,
+         
+         text_4_employment = case_when(abs(temp_working_diff) < 3 ~ "similar for children of immigrants and",
+                                       temp_working_diff < 6 & temp_working_diff >=3 ~ "slightly higher for children of immigrants than",
+                                       temp_working_diff >= 6 ~ "higher for children of immigrants than",
+                                       temp_working_diff > -6 & temp_working_diff <= -3 ~ "slightly lower for children of immigrants than",
+                                       temp_working_diff <= -6 ~ "lower for children of immigrants than"), 
+         
+         text_5_lowinc = case_when(temp_working_diff <= -3 | (share_above200povr_coi >- share_above200povr_nat) ~ "Around",
+                                   temp_working_diff > -3 ~ "Yet,",
+                                   TRUE ~ NA_character_),
+         
+         text_6_race = case_when((share_asian_coi >= 75 |
+                                  share_black_coi >= 75 | 
+                                  share_hispanic_coi >= 75 |
+                                  share_white_coi >= 75 ~ " fairly "),
+                                 TRUE ~ " "),
+         
+         temp_1_uninsure = (share_noinsure_all_0809) - (share_noinsure_all_1617),
+         
+         text_1_uninsure = case_when(temp_1_uninsure >= 1 ~ "fallen",
```

Comment:
Conditions 1 and 2 include all real numbers so I don't think condition 3 (`temp_1_uninsure <= 1 ~ "risen"`)` will ever be reached.

Reply:
`         text_1_uninsure = case_when(temp_1_uninsure >= 1 ~ "fallen"
                                     temp_1_uninsure > -1 ~ "not changed dramatically",
                                     TRUE ~ "risen" `
-- confirming this makes sense?

Reply:
resolved, I think